src/lib.rs

```diff
@@ -49,6 +49,45 @@ pub enum Error {
     InvalidId,
 }
 
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            #[cfg(feature = "remote_endpoint")]
+            Error::ChannelSend => write!(
+                f,
+                "event_manager: failed to send message to remote endpoint."
+            ),
+            #[cfg(feature = "remote_endpoint")]
+            Error::ChannelRecv => write!(
+                f,
+                "event_manager: failed to receive message from remote endpoint."
+            ),
+            #[cfg(feature = "remote_endpoint")]
+            Error::EventFd(_e) => write!(
+                f,
+                "event_manager: failed to manage EventFd file descriptor."
+            ),
+            Error::Epoll(_e) => write!(f, "event_manager: failed to manage epoll file descriptor."),
+            Error::FdAlreadyRegistered => write!(
+                f,
+                "event_manager: file descriptor has already been registered."
+            ),
+            Error::InvalidId => write!(f, "event_manager: invalid subscriber Id."),
+        }
+    }
+}
+
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            #[cfg(feature = "remote_endpoint")]
+            Error::EventFd(e) => Some(e),
+            Error::Epoll(e) => Some(e),
+            _ => None,
```

Comment:
I think we should explicitly enumerate the other variants here (instead of using the catch-all `_`). That way, we don't forget to update this function if we add a new variant to the `Error` enum.